src/ngx_srt_set_misc_module.c

```diff
@@ -0,0 +1,302 @@
+    block_size = EVP_CIPHER_block_size(cipher);
+    buf_size = input.len + block_size /* for EVP_DecryptUpdate */
+                       + block_size; /* for EVP_DecryptFinal_ex */
+
+    dst->len = buf_size;
+    dst->data = ngx_palloc(c->pool, buf_size);
+    if (dst->data == NULL) {
+        ngx_log_error(NGX_LOG_ERR, c->log, 0, "ngx_srt_set_misc_decrypt_aes:"
+            "alloc failed");
+        return NGX_ERROR;
+    }
+
+    ctx = EVP_CIPHER_CTX_new();
+
+    if (!EVP_DecryptInit_ex(ctx, cipher, NULL, key.data, iv.data))
+    {
+       ngx_log_error(NGX_LOG_ERR, c->log, 0, "ngx_srt_set_misc_decrypt_aes:"
+            "EVP_DecryptInit_ex failed");
+        EVP_CIPHER_CTX_cleanup(ctx);
+        return NGX_ERROR;
+    }
+
+    if (!EVP_DecryptUpdate(ctx, dst->data, (int *) &dst->len, input.data, input.len))
+    {
+       ngx_log_error(NGX_LOG_ERR, c->log, 0, "ngx_srt_set_misc_decrypt_aes:"
+            "EVP_DecryptUpdate failed");
+        EVP_CIPHER_CTX_cleanup(ctx);
+        return NGX_ERROR;
+    }
+
+    if (!EVP_DecryptFinal_ex(ctx, dst->data + dst->len, (int *) &dst->len))
+    {
+        ngx_log_error(NGX_LOG_ERR, c->log, 0, "ngx_srt_set_misc_decrypt_aes:"
+            "EVP_DecryptFinal_ex failed");
+        EVP_CIPHER_CTX_cleanup(ctx);
+        return NGX_ERROR;
+    }
+
+    EVP_CIPHER_CTX_cleanup(ctx);
+
+    return NGX_OK;
+}
+
+static ngx_int_t
+ngx_srt_set_misc_decrypt_variable(ngx_srt_session_t *s,
+    ngx_srt_variable_value_t *v, uintptr_t data)
+{
+    ngx_str_t               val, decrypt_str;
+    ngx_srt_encrypt_decrypt_ctx_t  *decrypt;
+
+    decrypt = (ngx_srt_encrypt_decrypt_ctx_t *) data;
+
+    ngx_log_debug0(NGX_LOG_DEBUG_STREAM, s->connection->log, 0,
+                   "srt decrypt started");
+
+    if (ngx_srt_complex_value(s, &decrypt->value, &val) != NGX_OK) {
+        ngx_log_error(NGX_LOG_ERR, s->connection->log, 0,
+        "ngx_srt_set_misc_decrypt_variable: failed to eval complex value");
+        return NGX_ERROR;
+    }
+
+    decrypt_str.len = val.len;
+    decrypt_str.data = val.data;
+
+    ngx_srt_set_misc_decrypt_aes(s->connection, decrypt->key, decrypt->iv,
+        val, &decrypt_str);
+
+    v->valid = 1;
+    v->no_cacheable = 0;
+    v->not_found = 0;
+    v->len = decrypt_str.len;
+    v->data = decrypt_str.data;
+
+    ngx_log_debug2(NGX_LOG_DEBUG_STREAM, s->connection->log, 0,
+                    "srt decrypt: \"%V\" \"%v\"", &val, v);
+
+    return NGX_OK;
+}
+
+static char *
+ngx_srt_set_misc_decrypt(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
+{
+    ngx_str_t                        *value, name;
+    ngx_srt_variable_t               *var;
+    ngx_srt_encrypt_decrypt_ctx_t    *decrypt;
+    ngx_srt_compile_complex_value_t   ccv;
+
+    decrypt = ngx_pcalloc(cf->pool, sizeof(ngx_srt_encrypt_decrypt_ctx_t));
+    if (decrypt == NULL) {
+        return NGX_CONF_ERROR;
+    }
+
+    value = cf->args->elts;
+
+    decrypt->key = value[1];
+    decrypt->iv  = value[2];
```

Comment:
since key & iv are binary, better have them base64 in nginx.conf and decode them here.
also - the validation of the binary size (iv.len == 16 & key.len = 16/24/32) better happen here, e.g. if i use key.len = 10, fail the config load instead of failing in runtime